### PR TITLE
PDJB-NONE: add isExpired accessor for JL invitation

### PR DESCRIPTION
**Pull request by samyou-softwire**
to be used for [PDJB-300](https://mhclgdigital.atlassian.net/browse/PDJB-300) and other tickets dealing with expired invitations

### CODE REVIEW

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitation.kt [hunk 1]
```diff
@@ -45,6 +50,18 @@ class JointLandlordInvitation(
...
+    val isExpired: Boolean
+        get() {
+            val dateTimeHelper = DateTimeHelper()
+
+            val expiresOnDate =
+                DateTimeHelper
+                    .getDateInUK(createdDate.toKotlinInstant())
+                    .plus(DatePeriod(days = JOINT_LANDLORD_INVITATION_LIFETIME_IN_DAYS))
+
+            return dateTimeHelper.getCurrentDateInUK() > expiresOnDate
```
Comment (TomHanmer29) [inline src/main/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitation.kt]: Just to check, is it expired ON the expiresOnDate or after the expiresOnDate? i.e. does it make sense to call this expiresOnDate and then use a >

**Review by TomHanmer29 — APPROVED** [review]
Just want to clarify one bit of logic/naming but otherwise seems sensible!

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitation.kt (hunk 1 above)
Comment (samyou-softwire) [reply src/main/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitation.kt]: good point, I'll double check what we want here with product and update accordingly

**Review by TomHanmer29 — CHANGES_REQUESTED** [review]
Just read Rowan's comments on https://github.com/communitiesuk/prsdb-webapp/pull/1402 and agree we should probably add some unit tests with edge cases for this :)

**Comment by samyou-softwire** [comment]
agree - product has clarified exactly how they want expiration to work so I'll get on this

**Comment by samyou-softwire** [comment]
added some tests @TomHanmer29, based on this from product (if the invitation expiry date was 06/06)

> "I think it should expire at midnight on 06/06 so that’d be the last day it’s valid"

File: src/test/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitationTests.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import uk.gov.communities.prsdb.webapp.constants.JOINT_LANDLORD_INVITATION_LIFETIME_IN_DAYS
+import uk.gov.communities.prsdb.webapp.testHelpers.mockObjects.MockJointLandlordData
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+class JointLandlordInvitationTests {
```
Comment (TomHanmer29) [inline src/test/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitationTests.kt:11]: Worth having a test specifically at midnight to ensure this is behaving as expected? Or is that overkill?

**Review by TomHanmer29 — APPROVED** [review]
Tiny comment on tests, but happy that this aligns with the product decision.

File: src/test/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitationTests.kt (hunk 2 above)
Comment (samyou-softwire) [reply src/test/kotlin/uk/gov/communities/prsdb/webapp/database/entity/JointLandlordInvitationTests.kt:11]: this would be a bit tricky to implement yeah, since we'd have to mock out the current time to test it expiring over the time boundary. I'll leave as it
